### PR TITLE
Gpl 626 location coordinates

### PR DESCRIPTION
**Pull request by Chris-Friend**
Closes #184 

Changes proposed in this pull request:

* Don't allow toggling coordinates on an existing Ordered or Unordered location: an UnorderedLocation should always not have coordinates; an Ordered always should ~~(note that we still need to address GPL-627 to ensure this latter criteria is fully met, e.g. coordinates can be set to (0, 0) which gives unexpected behaviour)~~ See comment below for behaviour

### CODE REVIEW

**Comment by Chris-Friend** [comment]
* On creating a location without the coordinated checkbox selected, or with 0 rows or 0 columns, an UnorderedLocation will be created
* On creating a location with the coordinate checkbox selected and rows > 0 and columns > 0, an OrderedLocation will be created
* As before there is no way to swap between UnorderedLocation and OrderedLocation types after creation
* On editing an existing UnorderedLocation, the coordinates checkbox is disabled and unchecked and therefore coordinates cannot be entered
* On editing an existing OrderedLocation, the coordinated checkbox is disabled and checked, and therefore coordinates can be entered. Note that (0, 0) can be entered, but this does not change the type of the location, nor does it prevent the coordinates from being updated to non-0 values at a later stage

File: app/views/locations/_form.html.erb [hunk 1]
```diff
@@ -50,17 +50,21 @@
...
+            <div class="table">
+              <div class="table-row">
+                <div class="table-head"><%= f.label :rows %></div>
+                <div class="table-head"><%= f.label :columns %></div>
+              </div>
+              <div class="table-row">
+                <div class="table-cell"><%= f.number_field :rows, min: 0 %></div>
+                <div class="table-cell"><%= f.number_field :columns, min: 0 %></div>
+              </div>
+            </div>
```
Comment (KatyTaylor) [inline app/views/locations/_form.html.erb:67]: Does this change the styling? Can I have a screenshot?

**Review by KatyTaylor — APPROVED** [review]

**Comment by KatyTaylor** [comment]
Looks good to me. The behaviour you mention in your final bullet point seems dodgy, so if you've got time you could fix that now too? Or if not, maybe make another bug / story for it?
"I should not be able to set coordinates to 0 for an OrderedLocation"

But maybe there's a use case for it. @stevieing what do you think?

**Comment by stevieing** [comment]
> Looks good to me. The behaviour you mention in your final bullet point seems dodgy, so if you've got time you could fix that now too? Or if not, maybe make another bug / story for it?
> "I should not be able to set coordinates to 0 for an OrderedLocation"
> 
> But maybe there's a use case for it. @stevieing what do you think?

I don't mind. If it is a quick fix go for it. If it is going to take a long time I would leave it as it is an edge case and it has never come up as an issue.

File: app/views/locations/_form.html.erb (hunk 1 above)
Comment (Chris-Friend) [reply app/views/locations/_form.html.erb:67]: LabwhereCoordinates

Comment (Chris-Friend) [reply app/views/locations/_form.html.erb:67]: It does change the styling. Happy to revert or try an alternative styling if you'd prefer

Comment (Chris-Friend) [reply app/views/locations/_form.html.erb:67]: In fact I think I'll revert back to the two list entries

Comment (Chris-Friend) [reply app/views/locations/_form.html.erb]: LabwhereCoordinatesNew

**Comment by Chris-Friend** [comment]
To confirm, I have change the behaviour so editing an OrderedLocation requires non-zero rows and columns inputs

File: app/views/locations/_form.html.erb (hunk 1 above)
Comment (KatyTaylor) [reply app/views/locations/_form.html.erb]: Not bothered either way, was just interested to see the result. Thanks :)

**Review by KatyTaylor — APPROVED** [review]

**Pull request merged**